ona/authentication.py

```diff
@@ -55,8 +57,15 @@ def authenticate(self, request):
         cookie_jwt = request.COOKIES.get(settings.KPI_COOKIE_NAME)
         if cookie_jwt:
             api_token = get_api_token(cookie_jwt)
-            if getattr(api_token, 'user'):
-                return api_token.user, api_token
+            user_name = api_token[0].user.username
+            user_email = api_token[0].user.email
+            # Create KPI User from Onadata user username and email
+            user = User.objects.using('default').get_or_create(
+                username=user_name, email=user_email)
+            if user:
+                # Grant permissions to KPI user
+                grant_default_model_level_perms(user)
+                return user, api_token
```

Comment:
Looking at this again, do we want to return `user` and `api_token` yet they are in different databases? For consistency, i'd say we return the `api_token` from the user that has been created. What do you think @DavisRayM @WinnyTroy ?

Reply:
I agree. _Is the `request.auth` used somewhere else for something ? If not we can just omit the `api_token` entirely_

Reply:
Looking at the onadata auth class, it seems we can return `(user, None)` as [here](https://github.com/onaio/onadata/blob/master/onadata/libs/authentication.py#L110) or `(user, True)` as seen [here](https://github.com/onaio/onadata/blob/master/onadata/libs/authentication.py#L258). I'd suggest we go we any of those 2

Reply:
The `TokenAuthentication` class we inherit from usually returns both the `(<auth_user>, <token>)`. I think it might be better to stick to that format for this unless `request.auth` value is not used anywhere specific ?

https://www.django-rest-framework.org/api-guide/authentication/#tokenauthentication
```
If successfully authenticated, TokenAuthentication provides the following credentials.

request.user will be a Django User instance.
request.auth will be a rest_framework.authtoken.models.Token instance.
```

Reply:
If we stick with that format, what token should we return? Do we or Django use the second tuple value - <token> - when it's returned? Technically, this is more of a [Custom authentication](https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication) which should inherit from `BaseAuthentication` and we can go ahead and borrow from [Ona's digest authentication class](https://github.com/onaio/onadata/blob/master/onadata/libs/authentication.py#L110) which returns `None` as the second tuple value. What do you guys think?

Reply:
Actually, in their [example](https://www.django-rest-framework.org/api-guide/authentication/#example), the second tuple value is `None`